app-backend/batch/src/main/scala/geojsonImport/ImportGeojsonFiles.scala

```diff
@@ -0,0 +1,170 @@
+        // access to, and this will let them view it even if they shouldn't be able to view it.
+        // I think we need a more formal system which requires signed uploads to specific directories, and
+        // reject any s3 urls that aren't in those directories like scene uploads
+        val s3Uri = new AmazonS3URI(URLDecoder.decode(uri, "utf-8"))
+        val s3Object = s3Client.getObject(s3Uri.getBucket, s3Uri.getKey)
+        val geojsonString =
+          new String(S3.getObjectBytes(s3Object), StandardCharsets.UTF_8)
+        logger.info("Annotations downloaded")
+        logger.info("Parsing annotations")
+        val annotations =
+          decode[AnnotationFeatureCollectionCreate](geojsonString) match {
+            case Right(fc) => fc.features.map(_.toAnnotationCreate).toList
+            case Left(e)   => throw e
+          }
+        logger.info("Annotations parsed")
+        annotations
+      }
+    }.flatten
+  }
+
+  def insertAnnotations(
+      annotations: List[Annotation.Create],
+      upload: GeojsonUpload
+  ): ConnectionIO[List[Annotation]] = {
+    // https://makk.es/blog/postgresql-parameter-limitation/
+    // max # of interpolated params in postgres driver = 32,767 (2 byte int)
+    // each annotation = 17 params
+    // 32,767 / 17 = 1927 annotations / batch. Cut that in half to 1000 to be safe.
+    for {
+      user <- UserDao.unsafeGetUserById(upload.createdBy)
+      inserted <- annotations
+        .grouped(1000)
+        .toList
+        .traverse(annotationBatch => {
+          val updatedAnnotationBatch =
+            annotationBatch
+              .map(_.copy(annotationGroup = Some(upload.annotationGroup)))
+          AnnotationDao.insertAnnotations(
+            updatedAnnotationBatch,
+            upload.projectId,
+            user,
+            Some(upload.projectLayerId)
+          )
+        })
+        .map { _.flatten }
+    } yield inserted
+  }
+
+  def main(args: Array[String]): Unit = {
+    val params = CommandLine.parser.parse(args, CommandLine.Params()) match {
+      case Some(params) =>
+        params
+      case None =>
+        logger.info("Invalid params")
+        return
+    }
+    val uploadId = params.upload
+    val uploadO = GeojsonUploadDao
+      .getUploadById(uploadId)
+      .transact(xa)
+      .unsafeRunSync()
+
+    uploadO match {
+      case Some(upload) =>
+        val annotations = processUploadToAnnotations(upload)
+        logger.info(
+          s"Uploading ${annotations.size} annotations to annotation group: ${upload.annotationGroup}"
+        )
+        params.testRun match {
+          case true =>
+            logger.info(
+              s"Test run: Not inserting ${annotations.size} annotations")
+          case _ =>
+            val inserted =
+              insertAnnotations(annotations, upload)
+                .transact(xa)
+                .unsafeRunSync()
```

Comment:
Same here -- pieces might fit together with extending `Job`, but we should try to keep the batch jobs declarative as much as we can.